src/device/wg_controller.rs

```diff
@@ -583,13 +603,35 @@ async fn build_requested_meshnet_peers_list<
                     // If peer is offline according to derp, we turn off keepalives.
                     None
                 } else {
+                    // Side effect: if peer is proxying but not marked offline by derp,
+                    // means it might want to upgrade to direct connection
+                    is_any_peer_proxying = true;
```

Comment:
I think we should have two conditions:
* Peer is proxying, just like you did here and peer has active WG session (last handshake < 180s)

Additionally I would recommend to move this pause/unpause logic to `consolidate_wg_peers()` logic, as generally `build_requested_peers_list()` might not be really a expected place to do actual state mutations, while `consolidate_wg_peers()` is designed to do exactly that :).